pip/tests/test_qsharp.py

```diff
@@ -32,6 +32,25 @@ def test_stdout_multiple_lines() -> None:
 
     assert f.getvalue() == "STATE:\n|0⟩: 1.0000+0.0000𝑖\nHello!\n"
 
+def test_dump_machine() -> None:
```

Comment:
@tcNickolas This test case shows how the API could be used to verify expected states between executions of qsharp snippets (these could also be `%%qsharp` magic cells with Python validation interspersed). Let me know what you think of the API, happy to iterate.